### PR TITLE
Project update

### PR DESCRIPTION
**Pull request by michalschott**
Tried to run this against kubernetes 1.12 but it complained about using custom transport.

Also included few tweaks.

Not sure if this project is still maintained?

### CODE REVIEW

**Review by szuecs — CHANGES_REQUESTED** [review]
This change will cause problems with DNS changes.
The custom transport is not a nice to have.

**Comment by michalschott** [comment]
@szuecs I've reverted the code in main.go and just disabled custom transport usage.

**Comment by szuecs** [comment]
The custom transport is required to make it work. 
Please show relevant logs, that show why you want to drop it. We might need to find another work around to get our transport into client-go.

There’s a client-go issue, that has to be solved before we can change this custom transport. Basically the fix is to rollout H2 within kubernetes client go and apiserver. Until it’s not done, we can not change this because of reliability.

File: Dockerfile [hunk 1]
```diff
@@ -1,7 +1,5 @@
-FROM registry.opensource.zalan.do/stups/alpine:latest
-MAINTAINER Team Teapot @ Zalando SE <team-teapot@zalando.de>
+FROM scratch
```
Comment (szuecs) [inline Dockerfile:1]: Why this change?
It doesn’t help in your issue and it’s not possible to investigate from inside containers anymore.

**Comment by michalschott** [comment]
```
kube-node-ready-5ff4bddfbf-577j6 kube-node-ready time="2019-08-05T12:24:57Z" level=fatal msg="using a custom transport with TLS certificate options or the insecure flag is not allowed"
```

That's the reason I've dropped custom transport completely.

> The custom transport is required to make it work.

Can you explain why? I'm open for better solution, altho this fix is currently working in my clusters w/o any problems so far.

File: Dockerfile (hunk 1 above)
Comment (michalschott) [reply Dockerfile:1]: I'll revert this once we agree on final solution, cherry-picked and squashed in rush

**Comment by szuecs** [comment]
You will rarely see the problem, but dns is only looked up once in Go. If the ip changes you have to delete the pod manually, which at our scale can’t do. https://github.com/golang/go/issues/23427 is the underlying Go issue and there’s also a kubernetes related client go issue https://github.com/kubernetes/client-go/issues/374#issuecomment-413923008 which leads to https://github.com/kubernetes/kubernetes/issues/65012
So you have to check if you can get rid of the fatal log without removing the whole custom transport as long H2 is not completely rolled out in k8s.
